packages/feedback/src/core/integration.ts

```diff
@@ -37,238 +37,256 @@ type Unsubscribe = () => void;
 
-    // START TEMP: Error messages
-    if (!modalIntegration && showScreenshot && !screenshotIntegration) {
-      throw new Error('Async loading of Feedback Modal & Screenshot integrations is not yet implemented');
-    } else if (!modalIntegration) {
-      throw new Error('Async loading of Feedback Modal is not yet implemented');
-    } else if (showScreenshot && !screenshotIntegration) {
-      throw new Error('Async loading of Feedback Screenshot integration is not yet implemented');
-    }
-    // END TEMP
+    /**
+     * Get the shadow root where we will append css
+     */
+    const _createShadow = (options: FeedbackInternalOptions): ShadowRoot => {
+      if (!_shadow) {
+        const host = DOCUMENT.createElement('div');
+        host.id = String(options.id);
+        DOCUMENT.body.appendChild(host);
 
-    if (!modalIntegration) {
-      // TODO: load modalIntegration
-      throw new Error('Not implemented yet');
-    }
+        _shadow = host.attachShadow({ mode: 'open' });
+        _shadow.appendChild(createMainStyles(options.colorScheme, options));
+      }
+      return _shadow as ShadowRoot;
+    };
 
-    if (showScreenshot && !screenshotIntegration && screenshotIsSupported) {
-      // TODO: load screenshotIntegration
-      throw new Error('Not implemented yet');
-    }
+    const _findIntegration = async <I extends Integration>(
+      integrationName: string,
+      getter: undefined | null | (() => IntegrationFn),
+      functionMethodName: 'feedbackModalIntegration' | 'feedbackScreenshotIntegration',
+    ): Promise<I> => {
+      const client = getClient();
+      const existing = client && client.getIntegrationByName(integrationName);
+      if (existing) {
+        return existing as I;
+      }
+      const integrationFn = (getter && getter()) || (await lazyLoadIntegration(functionMethodName));
+      const integration = integrationFn();
+      client && client.addIntegration(integration);
+      return integration as I;
+    };
 
-    return modalIntegration.createDialog({
-      options,
-      screenshotIntegration: screenshotIsSupported ? screenshotIntegration : undefined,
-      sendFeedback,
-      shadow: _createShadow(options),
-    });
-  };
+    const _loadAndRenderDialog = async (options: FeedbackInternalOptions): Promise<FeedbackDialog> => {
+      const [modalIntegration, screenshotIntegration] = await Promise.all([
+        _findIntegration<FeedbackModalIntegration>('FeedbackModal', getModalIntegration, 'feedbackModalIntegration'),
+        showScreenshot && isScreenshotSupported()
+          ? _findIntegration<FeedbackScreenshotIntegration>(
+              'FeedbackScreenshot',
+              getScreenshotIntegration,
+              'feedbackScreenshotIntegration',
+            )
+          : undefined,
+      ]);
+      if (!modalIntegration || (showScreenshot && !screenshotIntegration)) {
+        // trouble!
+        throw new Error('Missing feedback helper integration!');
```

Comment:
Do we know how an integration could be missing? Would be helpful to provide a potential solution or a link to one

Reply:
Idk if we strictly know what was supposed to be setup and failed. Certainly if either is missing then the async loading didn't work.

That could easily happen on purpose if the site relies on async loading but the user is offline right now, for example.

Reply:
I updated the comment with a TODO. We can for sure add help for the developer so they can make any adjustments, but user-facing i'm not sure where to go with it.